### PR TITLE
docs(madr): metrics

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**

### CODE REVIEW

File: docs/madr/001-metrics.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+* Latency of DB operations (exposed by Kuma)
+* Kuma API server responses
+* All Kuma components that have tickers (like insights, sub finalizer etc.)
+
+Number of
+* XDS reconciliations
+* Kubernetes reconciliations
+* Number of DB queries
+
+### Scenarios
```
Comment (Automaat) [inline docs/madr/001-metrics.md:38]: do we want to test how big clusters can be deployed without `reachable services`?

File: docs/madr/001-metrics.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+### Set of metrics
+
+P50, P90, P99 of
+* Time between applying Deployment and Pod be ready (exposed by Kubernetes?)
+* XDS config delivery (time between setting the config into snapshot up to receiving ACK/NACK)
+* XDS watchdog sync (time spent on generating XDS config)
+* Kubernetes reconciles (Pod to Dataplane conversion, VIP generation)
+* Latency of Kube API server responses (exposed by Kubernetes)
+* Latency of DB operations (exposed by Kuma)
+* Kuma API server responses
```
Comment (Automaat) [inline docs/madr/001-metrics.md:30]: we can have process that makes requests to cp to check response times from API

File: docs/madr/001-metrics.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+* All Kuma components that have tickers (like insights, sub finalizer etc.)
```
Comment (Automaat) [inline docs/madr/001-metrics.md:31]: we need to add this metrics

File: docs/madr/001-metrics.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+* Set of metrics
+
+## Decision Outcome
+
+Chosen option: "Set of metrics"
+
+## Pros and Cons of the Options
+
+### Set of metrics
```
Comment (Automaat) [inline docs/madr/001-metrics.md:21]: don't scrape DP metrics

File: docs/madr/001-metrics.md (hunk 1 above)
Comment (jakubdyszkiewicz) [reply docs/madr/001-metrics.md:45]: I don't think it make sense. We already know that it's not scalable solution

**Review by Automaat — APPROVED** [review]

File: docs/madr/001-metrics.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+P50, P90, P99 of
+* Time between applying Deployment and Pod be ready (exposed by Kubernetes?)
+* XDS config delivery (time between setting the config into snapshot up to receiving ACK/NACK)
+* XDS watchdog sync (time spent on generating XDS config)
+* Kubernetes reconciles (Pod to Dataplane conversion, VIP generation)
+* Latency of Kube API server responses (exposed by Kubernetes)
+* Latency of DB operations (exposed by Kuma)
+* Kuma API server responses
+  * We can create a process in the background to generate requests (like DP request every 1s) to see what is the latency.
+* All Kuma components that have tickers (like insights, sub finalizer etc.)
```
Comment (lobkovilya) [inline docs/madr/001-metrics.md:32]: What does it mean in terms of metrics? The caption says `P50, P90, P99 of` but this item does not quite follow the pattern

File: docs/madr/001-metrics.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  * It's not available yet.
+
+Number of
+* XDS reconciliations
```
Comment (lobkovilya) [inline docs/madr/001-metrics.md:36]: Is it the number of times the snapshot changed?

File: docs/madr/001-metrics.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+* Kubernetes reconciliations
+* Number of DB queries
```
Comment (lobkovilya) [inline docs/madr/001-metrics.md:38]: Should we count reads and writes separately?

File: docs/madr/001-metrics.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+* Set of metrics
+
+## Decision Outcome
+
+Chosen option: "Set of metrics"
+
+## Pros and Cons of the Options
+
+### Set of metrics
```
Comment (lobkovilya) [inline docs/madr/001-metrics.md:21]: please add mem and cpu utilization of the CP

File: docs/madr/001-metrics.md (hunk 5 above)
Comment (jakubdyszkiewicz) [reply docs/madr/001-metrics.md:32]: why? It's the latency of one "tick"

File: docs/madr/001-metrics.md (hunk 6 above)
Comment (jakubdyszkiewicz) [reply docs/madr/001-metrics.md:36]: no, it's a number of times we went through XDS generation

File: docs/madr/001-metrics.md (hunk 7 above)
Comment (jakubdyszkiewicz) [reply docs/madr/001-metrics.md:38]: we do already on the ResourceStore level. There is operation and resource type label

File: docs/madr/001-metrics.md (hunk 5 above)
Comment (lobkovilya) [reply docs/madr/001-metrics.md:32]: Just wasn't clear to me what we measure in these components. The latency of one "tick" makes sense

File: docs/madr/001-metrics.md (hunk 6 above)
Comment (lobkovilya) [reply docs/madr/001-metrics.md:36]: But we're going through XDS generation at a constant rate because of the SimpleWatchdog ticker, isn't it?

**Review by lobkovilya — APPROVED** [review]

**Pull request merged**